src/autoscaler/db/sqldb/scheduler_sqldb_test.go

```diff
@@ -123,4 +124,83 @@ var _ = Describe("SchedulerSqldb", func() {
 
 	})
 
+	Describe("SyncActivesSchedules", func() {
+		BeforeEach(func() {
+			sdb, err = NewSchedulerSQLDB(url, logger)
+			Expect(err).NotTo(HaveOccurred())
+			err = cleanSchedulerActiveScheduleTable()
+			Expect(err).NotTo(HaveOccurred())
+
+			err = insertSchedulerActiveSchedule(111111, "app-id-1", 1, 2, 10, 5)
+			Expect(err).NotTo(HaveOccurred())
+			err = insertSchedulerActiveSchedule(222222, "app-id-2", 2, 3, 7, 5)
+			Expect(err).NotTo(HaveOccurred())
+			err = insertSchedulerActiveSchedule(333333, "app-id-3", 3, 5, 12, -1)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			err = sdb.Close()
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		JustBeforeEach(func() {
+			schedules, err = sdb.GetActiveSchedules()
+		})
+		Context("when appIdMap is empty", func() {
+			BeforeEach(func() {
+				appIdMap = map[string]bool{}
+				err = sdb.SynchronizeActiveSchedules(appIdMap)
+				Expect(err).NotTo(HaveOccurred())
+			})
+			It("no active schedule is removed", func() {
+				Expect(schedules).To(HaveKeyWithValue("app-id-1", &models.ActiveSchedule{
+					ScheduleId:         "111111",
+					InstanceMin:        2,
+					InstanceMax:        10,
+					InstanceMinInitial: 5,
+				}))
+				Expect(schedules).To(HaveKeyWithValue("app-id-2", &models.ActiveSchedule{
+					ScheduleId:         "222222",
+					InstanceMin:        3,
+					InstanceMax:        7,
+					InstanceMinInitial: 5,
+				}))
+				Expect(schedules).To(HaveKeyWithValue("app-id-3", &models.ActiveSchedule{
+					ScheduleId:         "333333",
+					InstanceMin:        5,
+					InstanceMax:        12,
+					InstanceMinInitial: 0,
+				}))
+			})
+		})
+		Context("when appIdMap is not empty", func() {
+			BeforeEach(func() {
+				appIdMap = map[string]bool{
+					"app-id-2": true,
+				}
+				err = sdb.SynchronizeActiveSchedules(appIdMap)
+				Expect(err).NotTo(HaveOccurred())
+			})
+			It("active schedules of app which appId is not in appIdMap are removed", func() {
```

Comment:
Why not just check the length? It should be one and the only item should be for app-id-1.